doc/examples/applications/plot_3d_image_processing.py

```diff
@@ -0,0 +1,294 @@
+information was reported by the microscope used to image the cells.
+
+"""
+
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+import numpy as np
+from scipy import ndimage as ndi
+
+from skimage import (
+    exposure, io
+)
+from skimage.data import image_fetcher
+
+
+#####################################################################
+# Load and display 3D images
+# ==========================
+# Three-dimensional data can be loaded with `io.imread`.
+
+path = image_fetcher.fetch('data/cells.tif')
+data = io.imread(path)
+
+print("shape: {}".format(data.shape))
+print("dtype: {}".format(data.dtype))
+print("range: ({}, {})".format(data.min(), data.max()))
+
+# Report spacing from microscope
+original_spacing = np.array([0.2900000, 0.0650000, 0.0650000])
+
+# Account for downsampling of slices by 4
+rescaled_spacing = original_spacing * [1, 4, 4]
+
+# Normalize spacing so that pixels are a distance of 1 apart
+spacing = rescaled_spacing / rescaled_spacing[2]
+
+print("microscope spacing: {}\n".format(original_spacing))
+print("rescaled spacing: {} (after downsampling)\n".format(rescaled_spacing))
+print("normalized spacing: {}\n".format(spacing))
+
+#####################################################################
+# Let us try and visualize the (3D) image with `io.imshow`.
+
+try:
+    io.imshow(data, cmap="gray")
+except TypeError as e:
+    print(str(e))
+
+#####################################################################
+# Function `io.imshow` can only display grayscale and RGB(A) 2D images.
+# We can thus use it to visualize 2D planes. By fixing one axis, we can
+# observe three different views of the image.
+
+
+def show_plane(ax, plane, cmap="gray", title=None):
+    ax.imshow(plane, cmap=cmap)
+    ax.axis("off")
+
+    if title:
+        ax.set_title(title)
+
+
+_, (a, b, c) = plt.subplots(ncols=3, figsize=(15, 5))
+
+show_plane(a, data[32], title="Plane = 32")
+show_plane(b, data[:, 128, :], title="Row = 128")
+show_plane(c, data[:, :, 128], title="Column = 128")
+
+#####################################################################
+# As hinted before, a three-dimensional image can be viewed as a series of
+# two-dimensional planes. Let us write a helper function, `display`, to
+# display 30 planes of our data. By default, every other plane is displayed.
+
+
+def display(im3d, cmap="gray", step=2):
+    _, axes = plt.subplots(nrows=5, ncols=6, figsize=(16, 14))
+
+    vmin = im3d.min()
+    vmax = im3d.max()
+
+    for ax, image in zip(axes.flatten(), im3d[::step]):
+        ax.imshow(image, cmap=cmap, vmin=vmin, vmax=vmax)
+        ax.set_xticks([])
+        ax.set_yticks([])
+
+
+display(data)
+
+#####################################################################
+# Alternatively, we can explore these planes (slices) interactively using
+# Jupyter widgets. Let the user select which slice to display and show the
+# position of this slice in the 3D dataset.
```

Comment:
please add a sentence here to say that you need a Jupyter kernel to run for this to work, so people can open the notebook on binder or on their computer but it won't work in the static html page.

Reply:
Sure! Done 0e58bb1e -- did I even go into too many details?

Reply:
Looks great to me!